text/processing.py

```diff
@@ -171,15 +176,15 @@ def tasks(self):
 
     def run(self):
         if self.ratio:
-            assert not self.lines
+            assert not self.num_lines
```

Comment:
`self.num_lines` is not defined?

Reply:
This is because of `self.set_attrs(locals())`, so it is defined but PyCharm does not see that. I can change this, this is still legacy code.